### DISCUSSION TITLE
How to get the list of data which attribute to a certain point in the persistence diagram?

### DISCUSSION THREAD

**Discussion by Wookbarabara** [scikit-tda/persim#83] [Q&A] [answered]
Much thanks for sharing this wonderful work!

I am planning to use framework to deal with some experimental data. And I am interested in a certain point in the persistence diagram and want to know which data are contributing to this point at its brith. Is it possible to extract the positions of those data in the input data list? Or just to obtain an output list with those data storing inside?

**Comment by catanzaromj** [comment, marked as answer]
Hi @Wookbarabara. Thanks for the comment. 

Please take a look at the thread and comment posted here https://github.com/scikit-tda/ripser.py/discussions/168#discussioncomment-9975524. 

What you are looking for is referred to as a "representative cycle"--the thing that gives rise to the point in the persistence diagram. Because of speed and memory limitations, most modern persistent homology packages (like `ripser.py`) actually compute persistent cohomology and therefore, its easy to extract representative co-cycles. There's a notebook in the `ripser.py` docs that show how to do this.

If that isn't sufficient or you need more guidance, let me know.

**Reply by Wookbarabara** [reply]
Thank you for  sharing the solution! @catanzaromj 

I'm quite new to python, so it would take some time for me to learning this function. 
I will come back for further help if I cannot work it out :)

Much thanks to you!

**Reply by Wookbarabara** [reply]
Hi @catanzaromj . Sadly, I'm back with some questions.

I have tried to study the docs you provided, but there are several points I cannot fully understand yet.
I'm not quite familiar with the persistent cohomology theory, so I'm afraid my questions could look stupid.
Hoping you don't mind :)

Here come the questions:

1. The main idea to find the all data responsible for a "representative cycle" is that, we check the situation with a threshold slightly lower than the death time and situation with a threshold slightly higher than the birth time, to recognize the "cycle" which we are interested in in the persistence diagram? And then, we can find all data (points) which form the cycle. Is that right?

2.  At third graph in the docs (Representative Cocycles), what do the "1", "0" and "16" mean? if they are the No. of cocycles, then why cocycles between 1 and 16 are not shown?

3. What does it mean by being valid or not for a cocycle?

4. Does the value of coefficient field means that, when the sum of edges of a cocycle equal to this coefficient, then this cocycle is a valid cocycle? How can I understand the function of value of coefficient field? I have tried to change the value of "coeff" and the max blue number will change, but on other differences can be observed.

Sorry for asking so many basic questions. Please help me if it doesn't bother you too much!

Thank you very much!
